servo/connectors/kubernetes.py

```diff
@@ -2993,14 +3003,14 @@ async def rollback(self, error: Optional[Exception] = None) -> None:
 
     async def destroy(self, error: Optional[Exception] = None) -> None:
         """
-        Initiates the asynchronous deletion of the Deployment under optimization.
+        Initiates the asynchronous deletion of all pods in the Deployment under optimization.
```

Comment:
This makes sense if I think about it long enough but strikes me as being unintuitive and potentially surprising. When I configure a behavior called `destroy` and it doesn't destroy the target in the configuration context it would seem unintuitive.

What if we invert it by making the `on_failure` behavior for both the Deployment and Canary strategies called `shutdown` or `halt`? In my mind, having an abstract concept of shutting down/halting the strategy in whatever way is contextually appropriate makes deleting a Pod in one case and scaling to zero in another feel more natural.

This is my fault because I reached for symmetry with the `create` action but in hindsight that was a bad call because `destroy` is not a mandatory party of the object lifecycle like `create` is.

`on_failure: shutdown` or `on_failure: halt` both seem clear to me in either canary or deployment contexts where destroy doesn't.

The `Deployment` model operation can still remain `scale_to_zero` to keep that operation explicit. I may have just screwed up the naming of that failure mode enum value and strategy method from the start.

Reply:
I realized that part of my proposed changes may cause an issue with compatibility: replacing 'destroy' with 'shutdown' in the configuration definition will cause existing configs that use 'on_failure: destroy' to fail and require them to be updated.
If this could be an issue, we can allow 'destroy' alongside 'shutdown', but mark this as an obsolete feature, to have some time to replace all configurations out there.

Reply:
This is a good point. I agree with leaving support for `destroy` config value but we should log a warning if that value is provided. Then, in a later release, we can fully deprecate it by providing an explicit error message in a custom validator letting users know to update `destroy` to `shutdown`